### PR TITLE
Fix login-issue regarding cookie domain

### PR DESCRIPTION
**Pull request by ghophp**
fix the cookie domain problem removing the cookie for the subdomain every time that you access the url login.. the cookie for the respective domain continues intact..

### CODE REVIEW

File: app/com/yetu/oauth2provider/controllers/authentication/LoginPage.scala [hunk 1]
```diff
@@ -1,15 +1,38 @@
...
+          result = SecureSocial.withRefererAsOriginalUrl(
+            Ok(env.viewTemplates.getLoginPage(UsernamePasswordProvider.loginForm)))
+        }
+      }
+
+      // Avoid domain cookie interpolation
+      result.withHeaders(
+        "Set-Cookie" -> ("id=; domain=" +
+          StringUtils.subdomain(request.host) +
+          "; path=/; expires=Thu, 01 Jan 1970 00:00:00 GMT"))
```
Comment (jschaul) [inline app/com/yetu/oauth2provider/controllers/authentication/LoginPage.scala]: will this date format work for people with a browser configured in German, for instance?

Comment (ghophp) [inline app/com/yetu/oauth2provider/controllers/authentication/LoginPage.scala]: If the browser follow this RFC: http://tools.ietf.org/html/rfc6265#section-5.1.1 YES :D

Comment (jschaul) [inline app/com/yetu/oauth2provider/controllers/authentication/LoginPage.scala]: `result.withCookies(DiscardingCookie("id", domain=Some(StringUtils.subdomain(request.host)), secure=true).toCookie)` ?

File: test/com/yetu/oauth2provider/utils/StringUtilsSpec.scala [hunk 2]
```diff
@@ -50,4 +50,27 @@ class StringUtilsSpec extends BaseSpec {
...
+    "return the proper subdomain url from the given host without port" in {
+      val requestHost = "auth-dev.yetu.me"
+      StringUtils.subdomain(requestHost) mustBe ".yetu.me"
+    }
+    "return the proper subdomain url to 4 level subdomain" in {
+      val requestHost = "auth.prod.yetu.me"
+      StringUtils.subdomain(requestHost) mustBe ".yetu.me"
+    }
+  }
+
```
Comment (jschaul) [inline test/com/yetu/oauth2provider/utils/StringUtilsSpec.scala]: thanks for adding tests!

**Pull request merged**